astropy/table/tests/test_column.py

```diff
@@ -678,6 +677,25 @@ def test_col_unicode_sandwich_create_from_str(Column):
     assert np.all(c[:2] == np.array([uba, 'def']))
 
 
+@pytest.mark.parametrize('Column', (table.Column, table.MaskedColumn))
+def test_col_unicode_sandwich_bytes_obj(Column):
+    """
+    Create a Column of dtype object with bytestring in it and make sure
+    it keeps the bytestring and not convert to str with accessed.
+
+    See https://github.com/astropy/astropy/pull/8745
+    """
+    c = Column([None, b'def'])
+    assert c.dtype.char == 'O'
+    assert not c[0]
+    assert c[1] == 'def'  # Was b'def' before PR 8745
+    assert isinstance(c[1], str)  # Was bytes before PR 8745
+    assert not isinstance(c[1], bytes)  # Was str before PR 8745
```

Comment:
I noted the behavior change in comments here when compared to `astropy` 3.2.2. Is this behavior change acceptable?